stateproof/builder.go

```diff
@@ -165,11 +165,31 @@ func (spw *Worker) handleSig(sfa sigFromAddr, sender network.Peer) (network.Forw
 	spw.mu.Lock()
 	defer spw.mu.Unlock()
 
+	// The sig should be for a round which is a multiple of StateProofInterval
+	sfaBlkHdr, err := spw.ledger.BlockHdr(sfa.Round)
+	if err != nil {
+		// reject signatures for a round not in the ledger
+		// Should not disconnect this peer, since this is a fault of the relay
+		// The peer could have other signatures what the relay is interested in
+		return network.Ignore, err
+	}
+	proto := config.Consensus[sfaBlkHdr.CurrentProtocol]
+	if proto.StateProofInterval == 0 {
```

Comment:
later on- the builder check the following condition `sfa.Round < latestHdr.StateProofTracking[protocol.StateProofBasic].StateProofNextRound`
I think the condition you've added is contained by this one